.github/workflows/ci.yml

```diff
@@ -136,14 +136,94 @@ jobs:
         working-directory: crab-boil
         run: |
           ./criticalup link remove
-  
+
       - name: Run test cleanup workflow
         working-directory: crab-boil
         run: |
           ./criticalup remove
           ./criticalup clean
 
-          
+  integration-test-file-server:
+    name: Test CriticalUp-Dev against local file server on GHA Runners
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-latest
+    env:
+      CRITICALUP_DEV_TOKEN: ${{ secrets.CRITICALUP_DEV_TOKEN }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "Define toolchain"
+        shell: bash # Powershell does not do heredocs
+        run: |
+          mkdir -p crab-boil
+          cd crab-boil
+          cat <<- EOF > criticalup.toml
+            manifest-version = 1
+
+            [products.ferrocene]
+            release = "stable-25.02.0"
+            packages = [
+                "cargo-\${rustc-host}",
+                "rustc-\${rustc-host}",
+                "rust-std-\${rustc-host}",
+            ]
+          EOF
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: file-server
+          path: crab-boil
+          github-token: ${{ secrets.TEMPORARY_ACCESS_TOKEN_TO_AIR_GAPPED_ACTIONS_ARTIFACT }}
+          repository: ferrocene/criticalup-air-gapped-server
+          run-id: 16942640950
```

Comment:
The artifact only lives for 7 days right now, I'm not sure you want to merge this.